core/src/HIP/Kokkos_HIP_Space.cpp

```diff
@@ -176,23 +177,12 @@ Kokkos::HIP::allocation WARNING: The combination of device and system configurat
         }
 
         // check for correct runtime environment
-        const char* hsa_xnack = std::getenv("HSA_XNACK");
-        if (!hsa_xnack)
+        if (!Kokkos::Impl::is_xnack())
           std::cerr << R"warning(
-Kokkos::HIP::runtime WARNING: Kokkos did not find an environment variable 'HSA_XNACK'
-                              for the current process.
-                              Nevertheless, xnack is enabled for all processes if
-                              amdgpu.noretry=0 was set in the Linux kernel boot line.
+Kokkos::HIP::runtime WARNING: Kokkos was not able to verify that xnack is enabled. 
                               Without xnack enabled, Kokkos::HIPManaged might not behave
-                              as expected.)warning"
+                              as expected. Set HSA_XNACK=1 in your environment to enable it.)warning"
```

Comment:
Hmm we had the two staged warning/error because even if users set the `HSA_XNACK` variable correctly, the kernel has to support HMM. So technically setting the environment variable is not enough [see here](https://github.com/ROCm/ROCm/issues/2385#issuecomment-1874371854).
AFAIK the kernel will ignore command-line args it does not recognize. If that is correct, `amdgpu.noretry` can be present in the command-line even if HMM is missing in the kernel.